src/content/docs/references/docs/arrays.md

```diff
@@ -14,84 +14,132 @@ These are declared as `<type>[<size>]`, e.g. `int[4]`. Fixed arrays are treated
 
 Unlike C, fixed arrays do not decay into pointers. Instead, an `int[4]*` may be implicitly converted into an `int*`.
 
+```c3
+// C
+int foo(int *a) { ... }
 
-    // C
-    int foo(int *a) { ... }
-    
-    int x[3] = { 1, 2, 3 };
-    foo(x);
-    
-    // C3
-    fn int foo(int *a) { ... }
+int x[3] = { 1, 2, 3 };
+foo(x);
 
-    int[3] x = { 1, 2, 3 };
-    foo(&x);
+// C3
+fn int foo(int *a) { ... }
 
+int[3] x = { 1, 2, 3 };
+foo(&x);
+```
 
 When you want to initialize a fixed array without specifying the size, use the [*] array syntax:
+```c3
+int[3] a = { 1, 2, 3 };
+int[*] b = { 4, 5, 6 }; // Type inferred to be int[3]
+```
+## Slice
 
+The final type is the slice `<type>[]`  e.g. `int[]`. A slice is a view into either a fixed or variable array. Internally it is represented as a struct containing a pointer and a size. Both fixed and variable arrays may be converted into slices, and slices may be implicitly converted to pointers.
+
+```c3
+fn void test() 
+{
     int[3] a = { 1, 2, 3 };
-    int[*] b = { 4, 5, 6 }; // Type inferred to be int[3]
 
-## Slice
+    int[4]* b = (int[4]*)&a;    // Explicit cast required
```

Comment:
This example should be changed, it's under "slice" but shows no slices at all... The old one was better.

Reply:
I went through the conversion table to make some nice examples and documented it, hopefully that's better now, it helped my understanding too :)